examples/rl/pendulum/ex_save_and_load.py

```diff
@@ -0,0 +1,157 @@
+import functools
+import pickle
+import sys
+import time
+from datetime import datetime
+
+# Note: mujoco viewer must load before jax
+# isort: off
+import mujoco
+import mujoco.viewer
+
+# isort: on
+
+import jax
+from brax import envs
+from brax.io import model
+from brax.training import distribution
+from brax.training.acme import running_statistics
+from brax.training.agents.ppo import train as ppo
+from brax.training.agents.ppo.networks import make_inference_fn
+from mujoco import mjx
+
+from ambersim.rl.networks import MLP, BraxPPONetworksWrapper
+from ambersim.rl.pendulum.swingup import PendulumSwingupEnv
+
+"""
+Perform pendulum swingup training with custom policy architectures, and
+demonstrate saving and loading the trained networks from disk.
+"""
+
+
+def train_swingup():
+    """Train a pendulum swingup agent with custom network architectures."""
+    # Initialize the environment
+    envs.register_environment("pendulum_swingup", PendulumSwingupEnv)
+    env = envs.get_environment("pendulum_swingup")
+
+    # Create some funky custom network architectures
+    policy_network = MLP(layer_sizes=(512, 4, 128, 2))  # outputs mean and variance
+    value_network = MLP(layer_sizes=(26, 128, 53, 1))  # outputs a single value
+    network_wrapper = BraxPPONetworksWrapper(
+        policy_network=policy_network,
+        value_network=value_network,
+        action_distribution=distribution.NormalTanhDistribution,
+    )
+
+    train_fn = functools.partial(
+        ppo.train,
+        num_timesteps=100_000,
+        num_evals=50,
+        reward_scaling=0.1,
+        episode_length=200,
+        normalize_observations=True,
+        action_repeat=1,
+        unroll_length=10,
+        num_minibatches=32,
+        num_updates_per_batch=8,
+        discounting=0.97,
+        learning_rate=3e-4,
+        entropy_cost=0,
+        num_envs=1024,
+        batch_size=512,
+        network_factory=network_wrapper.make_ppo_networks,
+        seed=0,
+    )
```

Comment:
do you think there's any way to simplify this part at all? It would probably be the thinnest of wrappers so that we wouldn't have to use `partial` and instead just had a util called like `create_train_fn` or something like that so that we'd at least have a more descriptive util name. On the other hand, it might just bloat the package.

Reply:
I'm not opposed to wrapping this part, but it's still not obvious to me what of these parameters have clear defaults. My preference would be to wait until that's more obvious (e.g., implement a few more examples) to make that change in a separate PR

Reply:
ok, agreed.